src/actions/payoutMethods/schema.ts

```diff
@@ -0,0 +1,34 @@
+import { z } from 'zod';
+
+export const payoutMethodSchema = z.object({
+  upiId: z
+    .string()
+    .refine((value) => /^[0-9A-Za-z._-]{2,256}@[A-Za-z]{2,64}$/.test(value), {
+      message: 'Enter a valid UPI address',
+    })
+    .optional(),
+  solanaAddress: z
+    .string()
+    .refine((value) => /^[A-Za-z0-9]{44}$/.test(value), {
+      message: 'Enter a valid Solana address',
+    })
+    .optional(),
+});
+
+export const upiIdInsertSchema = z.object({
+  upiId: z
+    .string()
+    .refine((value) => /^[0-9A_Za-z._-]{2,256}@[A_Za-z]{2,64}$/.test(value), {
```

Comment:
Just little minor issue with the regix , i think its supposed to be [0-9A-Za-z._-]{2,256}@[A-Za-z]

Reply:
Thanks for catching that! I used A_Z instead of A-Z by mistake.I'll fix the regex to get it right.